### PR TITLE
[expo-av] Fix Audio.stopAndUnloadAsync not handling no-data on Android

### LINKED ISSUE #1709: Unable to stopAndUnloadAsync a recording that just got started
### Reproducible Demo
https://snack.expo.io/rJ6j67G6f

After calling `await recording.startAsync()`, it needs at least ~300ms before `recording.stopAndUnloadAsync()` can be called, otherwise it'll fail and subsequent `startAsync` calls will result in 
> error: Only one Recording object can be prepared at a given time.

Tested on Android.

### PR DESCRIPTION
**Pull request by IjzerenHein**
# Why

Fixes #1709, see https://github.com/expo/expo/issues/1709#issuecomment-678107702

# How

- Handle case when stopping recording while no audio data has been received
- Updated `stopAndUnloadAsync` to always clean-up/unload even if stop fails
- Added note to documentation
- Updated https://github.com/expo/audio-recording-example to demonstrate E_AUDIO_NODATA handling

# Test Plan

- Verified locally using audio-recording-example
- All test-suite tests succeed on Android

Screenshot_20200825-094854_Expo

### CODE REVIEW

**Comment by IjzerenHein** [comment]
Friendly ping to @bbarthec 😃

File: packages/expo-av/src/Audio/Recording.ts [hunk 1]
```diff
@@ -179,11 +179,10 @@ export class Recording {
 
   // Internal methods
 
-  _cleanupForUnloadedRecorder = async (finalStatus: RecordingStatus) => {
+  _cleanupForUnloadedRecorder = async (finalStatus?: RecordingStatus) => {
     this._canRecord = false;
     this._isDoneRecording = true;
-    // $FlowFixMe(greg): durationMillis is not always defined
-    this._finalDurationMillis = finalStatus.durationMillis;
+    this._finalDurationMillis = finalStatus?.durationMillis || 0;
```
Comment (bbarthec) [inline packages/expo-av/src/Audio/Recording.ts:185]:
```suggestion
    this._finalDurationMillis = finalStatus?.durationMillis ?? 0;
```

**Review by bbarthec — APPROVED** [review]
One minor suggestion 👍 
I'm sorry for not responding to this PR for this long 🙇

**Comment by IjzerenHein** [comment]
Great suggestion! I've updated the PR accordingly. Thanks for the review =D

**Review by bbarthec — APPROVED** [review]

**Comment by Aryk** [comment]
I'm encountering this issue on Android as well after stopping it quickly.

Are there any work arounds? It seems that once I get the error, there is no way for me clean things up again. I cannot create a new recording object. And even if I want 10 seconds, the stopAndUnloadAsync() call will never cleanup.

Is there anyway in a managed app to check that no data was received before calling stopAndUnloadAsync and putting my app into a broken state?

**Comment by Aryk** [comment]
Nevermind...I came up with a total hack...but seems to be working for now.

The real fix won't be out until SDK 40 right?


```ts
    try {
            await recordRef.current.stopAndUnloadAsync();
          } catch (e) {
            if (e.message.includes("Stop encountered an error: recording not stopped")) {
              await ExponentAV.unloadAudioRecorder();
              await recordRef.current._cleanupForUnloadedRecorder({durationMillis: 0} as any);
            } else {
              await handleError(e, {userMessage: "An error occurred stopping the recording."});
            }
          } finally {
            recordRef.current = undefined;
            timerStop();
            success = true;
          }
```

**Pull request merged**